### PR TITLE
lab 5 stringer

### LINKED ISSUE #305: Lab5

### PR DESCRIPTION
**Pull request by shuutech**
Fixes #305 

Review of colleague's PR #

### CODE REVIEW

File: 05_stringer/shuutech/main.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package main
+
+import (
+	"fmt"
+)
+
+type IPAddr [4]byte
+
+func (ipAddress IPAddr) String() string {
+	ip := fmt.Sprintf("%d.%d.%d.%d", ipAddress[0], ipAddress[1], ipAddress[2], ipAddress[3])
```
Comment (kasunfdo) [inline 05_stringer/shuutech/main.go:10]: Return directly without using a variable

File: 05_stringer/shuutech/main_test.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	var actual string
+	expected = "127.0.0.1"
+	actual = IPAddr{127, 0, 0, 1}.String()
+	if expected != actual {
+		t.Errorf("expected %v, got %v", actual, expected)
+		t.Fail()
+	}
+}
+
+func TestMain(m *testing.M) {
```
Comment (kasunfdo) [inline 05_stringer/shuutech/main_test.go:20]: This test will never fail. 

You can make main function testable using following printf method.
`fmt.Fprintln(out, IPAddr{127, 0, 0, 1})`

File: 05_stringer/shuutech/main_test.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import (
+	"fmt"
+	"os"
+	"testing"
+)
+
+func TestIpAddress(t *testing.T) {
+	var expected string
+	var actual string
+	expected = "127.0.0.1"
```
Comment (kasunfdo) [inline 05_stringer/shuutech/main_test.go:12]: Declare and assign directly as follows.
`expected := = "127.0.0.1"`

**Review by kasunfdo — CHANGES_REQUESTED** [review]
Implementation is OK, but testing should be improved.
Consider following.

- Uninitialized IPAddr(`IPAddr{}`)
- Semi-initialized IP Addr(`IPAddr{127,23}`)

**Comment by juliaogris** [comment]
The go-course is now closed. Thank you very much for participating.

**Pull request closed without merging**